processor/src/main/java/org/mapstruct/ap/internal/model/common/Type.java

```diff
@@ -758,10 +758,11 @@ public String getNull() {
             return "0";
         }
         if ( "char".equals( getName() ) ) {
-            return "'\\u0000'";
+            //"'\u0000'" would have been better, but depends on platform encoding
+                return "0";
         }
         if ( "double".equals( getName() ) ) {
-            return "0.0";
+            return "0.0d";
```

Comment:
I alligned this one with the default value that was use before. I did the same for the `char`.

We also need to decide whether we want to do it like we did before with the `TypeMirror` `Kind` or like it is here.